File: python/1679.max-number-of-k-sum-pairs.py
class Solution:
    def maxOperations(self, nums: List[int], k: int) -> int:
        m = defaultdict(int)
        res = 0
        for i in nums:
            if m[k - i] > 0:
                res += 1
                m[k - i] -= 1
            else:
                m[i] += 1
        
        return res

File: golang/1679.max-number-of-k-sum-pairs.go
func maxOperations(nums []int, k int) int {

  sort.Ints(nums)

  op := 0

  for i, j := 0, len(nums) - 1; i < j; {
    s := nums[i] + nums[j]

    if s < k {
      i++
    } else if s > k {
      j--
    } else {
      op++
      i++
      j--
    }

  }

  return op
    
}